### PR TITLE
feat: add hideTags option in user settings

### LINKED ISSUE #3350: Hide tags from detail page
### Description

The ability to hide tags from the detail page

Apart from it not being potentially useful to some users and causing visual clutter (especially on mobile screens), it also gives away spoilers for the content.

image

### Desired Behavior

The ability to hide tags

### Additional Context

_No response_

### Code of Conduct

- [X] I agree to follow Overseerr's Code of Conduct

### PR DESCRIPTION
**Pull request by andromidasj**
#### Description
This new option will allow for a user to hide the tags that show in a Movie or TV Show details page. If the setting `Hide Tags` is checked, the tags will not show. If it is unchecked, the tags will show.

#### Screenshot (if UI-related)
Setting:
image

Before:
image

After:
image

#### To-Dos

- [x] Successful build `yarn build`
- [x] Translation keys `yarn i18n:extract`
- [ ] Database migration (if required)

#### Issues Fixed or Closed

- Fixes #3350

### CODE REVIEW

File: src/hooks/useUser.ts [hunk 1]
```diff
@@ -25,15 +26,14 @@ export interface User {
 
 type NotificationAgentTypes = Record<NotificationAgentKey, number>;
 
-export interface UserSettings {
-  discordId?: string;
-  region?: string;
-  originalLanguage?: string;
-  locale?: string;
+type UserSettings = Pick<
```
Comment (andromidasj) [inline src/hooks/useUser.ts:29]: I noticed here that we're duplicating code and having multiple sources of truth. This Pick type allows us to honor the types already set for these settings in the UserSettingsGeneralResponse interface. If it's desired to have it typed out explicitly in this file though, we can simply remove what I added and instead add `hideTags?: boolean;` underneath line 32 of the original code.

File: src/hooks/useUser.ts [hunk 2]
```diff
@@ -25,15 +26,14 @@ export interface User {
...
+  UserSettingsGeneralResponse,
+  'discordId' | 'region' | 'originalLanguage' | 'locale' | 'hideTags'
+> & {
   notificationTypes: Partial<NotificationAgentTypes>;
   watchlistSyncMovies?: boolean;
   watchlistSyncTv?: boolean;
-}
+};
```
Comment (andromidasj) [inline src/hooks/useUser.ts:36]: I noticed here that we're duplicating code and having multiple sources of truth. This Pick type allows us to honor the types already set for these settings in the UserSettingsGeneralResponse interface. If it's desired to have it typed out explicitly in this file though, we can simply remove what I added and instead add `hideTags?: boolean;` underneath line 32 of the original code.

Comment (OwsleyJr) [reply src/hooks/useUser.ts:36]: I don't think we should modify any older code unless necessary. The hideTags boolean should suffice.

Comment (andromidasj) [reply src/hooks/useUser.ts:36]: Sounds good, I'll remove this chunk then!

**Comment by andromidasj** [comment]
@OwsleyJr do you know how I can get help with data migration stuff? My understanding is pretty limited in that side of things, and there seems to be some SQL build errors in Cypress like:

```
driverError: Error: SQLITE_ERROR: no such column: User_settings.hideTags
```

**Comment by OwsleyJr** [comment]
> @OwsleyJr do you know how I can get help with data migration stuff? My understanding is pretty limited in that side of things, and there seems to be some SQL build errors in Cypress like:
> 
> ```
> driverError: Error: SQLITE_ERROR: no such column: User_settings.hideTags
> ```

Would just need to generate a migration file. You can look in the package.json for the script we use.

**Pull request closed without merging**